common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -493,6 +493,11 @@ public boolean inEventLoop(Thread thread) {
         return thread == this.thread;
     }
 
+    @Override
+    public boolean inEventLoop() {
+        return Thread.currentThread() == nonVolatileThread;
```

Comment:
@njhill so we depend on the java memory model to basically have nonVolatileThread be made correctly visible by other synchronised / volatile ops before ?

Reply:
@normanmaurer I should add some comment here as I agree it looks deceptively unsafe, but I'm reasonably confident that this is robust:
- If we are the event loop thread, it will clearly return true since the field is only written by that thread.
- If we are not the event loop thread, there's no way we can observe this value as equal to us since it will either have never been set equal to us, or if it ever was in the past (i.e. a terminated event loop for which we were previously the thread), we would have been guaranteed to subsequently set it to null prior to reading this.

It's possible I may have missed something though given the [warnings](http://gee.cs.oswego.edu/dl/html/j9mm.html#plainsec) against reasoning about correctness of concurrent plain mode access. So would be good to get this blessed by one of the concurrency gurus.

Reply:
The reasoning of Nick is quite robust here :+1: 
Plane mode access could be complex here only if we rely on some ordered effect due to the outcome of this check, but if not (and it seems the case) I cannot see anything wrong in this optimization, although we show some love of running with scissors here XD

Reply:
@franz1981 @njhill sounds about right...

@njhill can you add a comment ?

Reply:
Thanks @franz1981! And I ack your point about measurable benefits, will do some benchmarking tomorrow :)

Reply:
@normanmaurer have now added the comment. Will get to the benchmarks at some point...